react/components/AppVersionContext.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useContext, createContext } from 'react'
+import { useRuntime, withRuntimeContext } from 'vtex.render-runtime'
+import { compose, graphql } from 'react-apollo'
+
+import * as appMajorsQuery from '../graphql/getAppMajors.graphql'
+
+type Dispatch = (newMajorInfo: {
+  major: string
+  availableMajors: string[]
+}) => void
+interface State {
+  major: string
+  availableMajors: string[]
+}
+
+const AppVersionStateContext = createContext<State | undefined>(undefined)
+const AppVersionDispatchContext = createContext<Dispatch | undefined>(undefined)
+
+function AppVersionProvider({ children, appMajorsQuery }: any) {
+  const {
+    route: { params },
+  } = useRuntime()
+  const urlVersion = params.app.split('@')[1]
+  const hasVersion = !!urlVersion
+
+  const [versionInfo, setVersionInfo] = useState({
+    major: `${
+      hasVersion ? urlVersion : appMajorsQuery.getAppMajors.latestMajor
```

Comment:
What happens if appMajorsQuery is loading?

Reply:
Good question hahaha should handle that